test/acceptance/belongs_to_required_test.exs

```diff
@@ -0,0 +1,202 @@
+defmodule Test.Acceptance.BelongsToRequiredTest do
+  use ExUnit.Case, async: true
+
+  defmodule Author do
+    use Ash.Resource,
+      data_layer: Ash.DataLayer.Ets,
+      extensions: [
+        AshJsonApi.Resource
+      ]
+
+    ets do
+      private?(true)
+    end
+
+    json_api do
+      type("author")
+
+      routes do
+        base("/authors")
+        get(:default)
+        index(:default)
+      end
+
+      fields [:id, :name]
+    end
+
+    actions do
+      read(:default)
+
+      create(:default)
+    end
+
+    attributes do
+      attribute(:id, :uuid, primary_key?: true)
+      attribute(:name, :string)
+    end
+
+    relationships do
+      has_many(:posts, Test.Acceptance.BelongsToRequiredTest.Post, destination_field: :author_id)
+    end
+  end
+
+  defmodule Post do
+    use Ash.Resource,
+      data_layer: Ash.DataLayer.Ets,
+      extensions: [
+        AshJsonApi.Resource
+      ]
+
+    ets do
+      private?(true)
+    end
+
+    json_api do
+      type("post")
+
+      routes do
+        base("/posts")
+        get(:default)
+        index(:default)
+        post(:default)
+      end
+
+      fields [:id, :name, :hidden, :email, :author]
+    end
+
+    actions do
+      read(:default)
+
+      create :default do
+        accept([:id, :name, :hidden, :author])
+      end
+    end
+
+    attributes do
+      attribute(:id, :uuid, primary_key?: true, allow_nil?: false)
+      attribute(:name, :string, allow_nil?: false)
+      attribute(:hidden, :string)
+
+      attribute(:email, :string,
+        allow_nil?: true,
+        constraints: [
+          match: ~r/[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}/
+        ]
+      )
+    end
+
+    relationships do
+      belongs_to(:author, Test.Acceptance.BelongsToRequiredTest.Author, required?: true)
+    end
+  end
+
+  defmodule Api do
+    use Ash.Api,
+      extensions: [
+        AshJsonApi.Api
+      ]
+
+    json_api do
+      log_errors?(false)
+    end
+
+    resources do
+      resource(Post)
+      resource(Author)
+    end
+  end
+
+  import AshJsonApi.Test
+
+  @tag :attributes
+  describe "invalid_post" do
+    test "create without all author_id in relationship" do
+      id = Ecto.UUID.generate()
+
+      response =
+        Api
+        |> post("/posts", %{
+          data: %{
+            type: "post",
+            attributes: %{
+              id: id,
+              name: "Invalid Post 1"
+            },
+            relationships: %{
+              author: %{
```

Comment:
Try this with
```
author: nil
```

This request should actually have been rejected as invalid.

Reply:
Have tried that, for current code and `nil` , response is same.
```
resp_body: %{
    "errors" => [
      %{
        "code" => "FrameworkError",
        "detail" => "Returned when an unexpected error in the framework has occured.",
        "id" => "c8fe0c27-6cff-46c1-8761-d06d3c1fc042",
        "title" => "Framework Error"
      }
    ],
    "jsonapi" => %{"version" => "1.0"}
  },
```

Reply:
I'll have to solve that problem before we merge this then, but aside from that everything looks good.

Reply:
ok

Reply:
Let me know if i can do anything to help.